z3/src/solver.rs

```diff
@@ -213,6 +223,19 @@ impl<'ctx> Solver<'ctx> {
         })
     }
 
+    /// Return a brief justification for an "unknown" result (i.e., `LiftedBool::Undefined`) for
+    /// the commands [`Solver::check()`](#method.check) and
+    /// [`Solver::check_assumptions()`](#method.check_assumptions)
+    pub fn get_reason_unknown(&self) -> String {
```

Comment:
Please use the same code structure as we do in the various `Display` impls so that it matches those in terms of the creation of the string, checking for NULL, etc.

Reply:
I opted to return an `Option<String>`.